src/main/java/org/spongepowered/api/entity/player/tab/PlayerTabInfo.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.player.tab;
+
+import org.spongepowered.api.GameProfile;
+import org.spongepowered.api.entity.player.gamemode.GameMode;
+import org.spongepowered.api.text.message.Message;
+
+/**
+ * Represents the information attached to a player in a {@link TabList}.
+ */
+public interface PlayerTabInfo {
+
+    /**
+     * Gets the connection time for this player.
+     * 
+     * @return The connection time for this player
+     */
+    int getConnectionTime();
+
+    /**
+     * <p>Sets the connection time for this player. The client displays
+     * connection bars based on this number.</p> <table summary=""> <thead>
+     * <tr><th>Bars</th><th>Time</th></tr> </thead> <tbody> <tr><td>0</td><td>
+     * Less than 0 </td></tr> <tr><td>1</td><td> 1000+ </td></tr>
+     * <tr><td>2</td><td> 600 - 999</td></tr> <tr><td>3</td><td> 300 -
+     * 599</td></tr> <tr><td>4</td><td> 150 - 299 </td></tr> <tr><td>5</td><td>
+     * 0 - 149 </td></tr> </tbody> </table>
+     * 
+     * @param milliseconds The new connection time, in milliseconds
+     */
+    void setConnectionTime(int milliseconds);
+
+    /**
+     * Gets the {@link GameMode} this player is in.
+     * 
+     * @return The Gamemode this player is in
+     */
+    GameMode getGameMode();
+
+    /**
+     * Sets this player's gamemode. Note that setting this to spectator is what
+     * enables spectator (noclip, invisibility).
+     * 
+     * @param gamemode The new gamemode
+     */
+    void setGameMode(GameMode gamemode);
+
+    /**
+     * Gets this player's display name.
+     * 
+     * @return This player's display nam
+     */
+    Message getDisplayName();
+
+    /**
+     * Sets this player's display name.
+     * 
+     * @param displayName The new display name
+     */
+    void setDisplayName(Message displayName);
+
+    /**
+     * Gets this player's username.
+     * 
+     * @return This player's username
+     */
+    String getName();
+
+    /**
+     * Sets this player's username.
+     * 
+     * @param name A new username
+     */
+    void setName(String name);
+
+    /**
+     * Gets the {@link GameProfile} associated with this player.
+     * 
+     * @return The GameProfile associated with this player
+     */
+    GameProfile getProfile();
+
+    /**
+     * Sets the {@link GameProfile} associated with this player.
+     * 
+     * @param profile The new GameProfile for this player
+     */
+    void setProfile(GameProfile profile);
```

Comment:
May be worth noting that in the javadoc. It's not immediately obvious (to me anyway)